data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,844 @@
+  armor_class : int;
+}
+```
+Values of type `character` carry the same data as inhabitants of this product: `string * int * string * character_class * character_alignment * int`.
+
+Access to the fields is done using the dot notation. Here is an example:
+```ocaml
+# let ghorghor_bey = {
+    name = "Ghôrghôr Bey";
+    level = 17;
+    race = "half-ogre";
+    class_type = Fighter;
+    alignment = (Chaotic, R_Neutral);
+    armor_class = -8;
+  };;
+val ghorghor_bey : character =
+  {name = "Ghôrghôr Bey"; level = 17; race = "half-ogre";
+   class_type = Fighter; alignment = (Chaotic, R_Neutral); armor_class = -8}
+
+# ghorghor_bey.alignment;;
+- : firmness * rectitude = (Chaotic, R_Neutral)
+
+# ghorghor_bey.class_type;;
+- : character_class = Fighter
+
+# ghorghor_bey.level;;
+- : int = 17
+```
+
+To some extent, records also are variants, with a single constructor carrying all the fields as a tuple. Here is how to alternatively define the `character` record as a variant.
+```ocaml
+# type character' = Character of string * int * string * character_class * (firmness * rectitude) * int;;
+type character' =
+    Character of string * int * string * character_class *
+      (firmness * rectitude) * int
+
+# let name (Character (name, _, _, _, _, _)) = name;;
+val name : character' -> string = <fun>
+
+# let level (Character (_, level, _, _, _, _)) = level;;
+val level : character' -> int = <fun>
+
+# let race (Character (_, _, race, _, _, _)) = race;;
+val race : character' -> string = <fun>
+
+# let class_type (Character (_, _, _, class_type, _, _)) = class_type;;
+val class_type : character' -> character_class = <fun>
+
+# let alignment (Character (_, _, _, _, alignment, _)) = alignment;;
+val alignment : character' -> firmness * rectitude = <fun>
+
+# let armor_class (Character (_, _, _, _, _, armor_class)) = armor_class;;
+val armor_class : character -> int = <fun>
+```
+
+One function will retrieve and read the contained data per field. It provides the same functionality as the dotted notation.
+```ocaml
+# let ghorghor_bey' = Character ("Ghôrghôr Bey", 17, "half-ogre", Fighter, (Chaotic, R_Neutral), -8);;
+val ghorghor_bey' : character =
+  Character ("Ghôrghôr Bey", 17, "half-ogre", Fighter, (Chaotic, R_Neutral), -8)
+
+# level ghorghor_bey';;
+- : int = 17
+```
+Writing `level ghorghor_bey'` is the same as `ghorghor_bey.level`.
+
+**Remarks**
+
+1. To be true to facts, it is not possible to encode all records as variants because OCaml provides a means to define fields whose value can be updated, which isn't available while defining variant types. This will be detailed in an upcoming tutorial on imperative programming.
+
+1. Records **should not** be defined using this technique. It is only demonstrated here to further illustrate the expressive strength of OCaml variants.
+
+1. This way, to define records **may** be applied to _Generalised Algebraic Data Types_, which is the subject of another tutorial.
```

Comment:
To define record may be applied to...
Defining records may be applied to...
The definition of records may be applied to...

Which of these is correct?

Reply:
So like: "Defining records in this way may be applied to ...."
"This method of defining records may be applied to..."

In English we'd say something more like: "This is the way to do the hula hoop" or "The way to do the hula hoop is to rotate your hips...."

Reply:
I understand, thanks for the clarification. I like 
> Defining records in this way may be applied to